src/cli.rs

```diff
@@ -1,17 +1,65 @@
 use bfint::interpret;
-use std::io::{self, Read};
+use std::fs;
+use std::io::{self};
 
-fn get_code_from_stdin() -> io::Result<Vec<u8>> {
-    let mut buf: Vec<u8> = vec![];
-    io::stdin().read_to_end(&mut buf).expect("aboba");
-    Ok(buf)
+use clap::{Args, Parser, Subcommand};
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+#[command(propagate_version = true)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Run brainfuck code (--name file, --source str)
+    Run(Run),
+}
+
+#[derive(Args)]
+struct Run {
+    #[command(flatten)]
+    src_from: SrcFrom,
+}
+
+#[derive(Args, Debug)]
+#[group(required = true, multiple = false)]
+struct SrcFrom {
+    /// Brainfuck source code
+    #[arg()]
+    arg: Option<String>,
+    /// Path to file with brainfuck code
+    #[arg(short, long)]
+    file: Option<String>,
 }
 
 fn main() {
-    let code = match get_code_from_stdin() {
-        Ok(code) => code,
-        Err(_) => panic!("failed to parse code"),
-    };
+    let cli = Cli::parse();
+
+    // You can check for the existence of subcommands, and if found use their
+    // matches just as you would the top level cmd
+    match &cli.command {
+        Commands::Run(run_args) => {
+            if let Some(source_code) = &run_args.src_from.arg {
+                interpret(
+                    source_code.as_bytes().to_vec(),
+                    &mut io::stdout(),
+                    &mut io::stdin(),
+                )
+            }
 
-    interpret(code, &mut io::stdout(), &mut io::stdin())
+            if let Some(file_path) = &run_args.src_from.file {
+                match fs::read(file_path) {
+                    Ok(source_code) => interpret(source_code, &mut io::stdout(), &mut io::stdin()),
+                    Err(err) => {
+                        println!("Cannot read file with path: '{}'", file_path);
```

Comment:
```suggestion
                        println!("Cannot read file with path: '{}': {}", file_path, err);
```
error can be written here

Reply:
Then it's better like this: 
```rust
println!("Cannot read file with path: '{}': erorr: '{}'", file_path, err);
```

Reply:
too many ":" imho